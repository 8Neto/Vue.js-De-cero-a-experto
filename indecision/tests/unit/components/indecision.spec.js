/**
* @vitest-environment happy-dom
*/

import { mount  } from '@vue/test-utils'
import { beforeEach, describe, expect, it, vi } from 'vitest'
import Indecision from '@/components/Indecision.vue'



describe('Indecision Component', ()=>{

    let wrapper
    let clgSpy

    beforeEach(()=>{
        wrapper = mount(Indecision)

        clgSpy = vi.spyOn(console, 'log')

    })

    it('debe de hacer match con el snapshot', () =>{
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('Escribir en el input no debe disparar nada (console.log)', async () =>{
        const getAnswerSpy = vi.spyOn( wrapper.vm, 'getAnswer' )

        const input = wrapper.find('input')

        await input.setValue('Hola Mundo')


        expect( clgSpy ).toHaveBeenCalledTimes(1)
        expect( getAnswerSpy ).not.toHaveBeenCalled()
    })

    it('Escribir en "?" debe disparar el getAnswer', async () =>{

        const getAnswerSpy = vi.spyOn( wrapper.vm, 'getAnswer' )

        const input = wrapper.find('input')

        await input.setValue('Hola mundo?')

        expect( getAnswerSpy ).toHaveBeenCalledTimes(1)
    })

    it('Pruebas en getAnswer', () =>{
        // expect().toMatchSnapshot()
    })

    it('Pruebas en getAnswer - Fallo en el API', () =>{
        // expect().toMatchSnapshot()
    })

})
