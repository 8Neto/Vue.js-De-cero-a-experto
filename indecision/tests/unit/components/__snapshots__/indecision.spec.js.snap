// Vitest Snapshot v1

exports[`Indecision Component > debe de hacer match con el snapshot 1`] = `
"<!--v-if-->
<div class=\\"bg-dark\\" data-v-4a8fca50=\\"\\"></div>
<div class=\\"indecision-container\\" data-v-4a8fca50=\\"\\"><input type=\\"text\\" placeholder=\\"Hazme una pregunta\\" data-v-4a8fca50=\\"\\">
  <p data-v-4a8fca50=\\"\\">Recuerda terminar con un signo de interrogaci√≥n (?)</p>
  <!--v-if-->
</div>"
`;
